### ISSUE TITLE
The Deserts of the West: A travel guide to unknown lands

### ISSUE THREAD

**Issue by mewo2** [dariusk/NaNoGenMo-2015#156]
In the grand tradition of fantasy novelists, I've got very bogged down in world-building, and particularly in generating maps. So I've decided that I'm going to make a guidebook of sorts, with maps of distant lands, and descriptions of the wonders that can be found there.

Some test maps:

![A test map](https://cloud.githubusercontent.com/assets/360952/11129638/4c1a1bc2-8978-11e5-9698-639bc4c9babd.png)

![Prop Prop Prop](https://cloud.githubusercontent.com/assets/360952/11129683/adf7b2d2-8978-11e5-9015-fa63c71f7b1d.png)

**Comment by hugovk** [comment]
These look great!

**Comment by maetl** [comment]
:heart:

**Comment by mewo2** [comment]
I've been playing around with some text generation. It started out as an Invisible Cities pastiche, but it's gone in a slightly different direction.

![map](https://cloud.githubusercontent.com/assets/360952/11171437/6b8cbc9a-8be8-11e5-82c2-654e8ffaba4d.png)

# Shurgulz

On entering the city, one may catch sight of the colossal castle of
Barturk. From here, a traveller cannot discern how merchants crowd
around its gate. Slightly closer to the heart of the city, the traveller
will begin to hear songs of dirgeful ecstasy, sung by the bourgeoisie of
the city. In the end the fragrance of _ritig_ fruit fills the wind, and
one has undeniably arrived.

In the Ancient Quarter, the alleyways are lined with _lirbsha_ plants.
Philosophers flock here, practising their skill at _daplar_. In the
event that a traveller is so lucky as to stumble upon an argument
between two lovers, it may be a humbling experience. The caged _nanil_
birds will be obvious to one. These denote the home of an artisan. In
the afternoon, the traveller can sometimes hear the song of the _raski_
birds.

# Kirrab

From Shurgulz a traveller may travel west to Kirrab. It is a pleasant
journey. As the traveller enters Zhiturk, the greenery changes, giving
way to open fields and thickets of _lirbra_ flowers. At one point the
route crosses a deep gorge. Near the roadside, mighty _nirshuch_ graze.

Near to the monumental city of Kirrab, one will come across a _nirpi_,
prized for its horns. Somewhat nearer to the centre of the city, the
traveller will hear the song of the _raski_ birds. At last the musk of
_biltarb_ trees fills the air, and one is beyond doubt in Kirrab.

In the Grey District, the boulevards are paved with lead, inlaid with
obsidian. A traveller may usually hear the sound of the prison clock.
The traveller will take note of the _raltarb_ flowers. These are a
warning to evil spirits.

# Rutig

Leaving Kirrab a traveller can journey north to Rutig. It is an
enjoyable journey. Along the way the track crosses a broad valley. When
the traveller crosses into Dirchals, there is a variation in the light,
and a harsh cast falls over the terrain. By the verge of the road,
_lirbra_ trees grow.

Entering famed Rutig, a traveller can encounter a series of pedlars,
setting out their stalls. A little nearer to the inner parts of the
city, a traveller will hear songs of plaintive joy, sung by citizens of
Rutig. Eventually the bouquet of _chalsri_ fruit fills the wind, and one
is surely in Rutig.

Around the citadel, the squares are paved with granite, inlaid with
limestone. Writers mill here, praying. Should the traveller be
unfortunate enough to see artisans talking, one may be drawn in, and
unable to leave. The caged _nanil_ birds will be clear. These serve to
warn off unwanted influences.

# Gulztig

From Rutig one may travel to Gulztig. The route is well-trodden, and the
journey takes but an afternoon. When a traveller crosses into Pibsir,
the foliage changes, giving way to hedgerows and _biltarb_ plants. On
the track, _raltarb_ plants grow. At one point the road crosses a deep
gorge, spotted with _shapi_ flowers.

Outside the tremendous city of Gulztig, the traveller will espy the
sandstone lookout of the gargantuan fortress of Shurti. A little closer
to the inner parts of the city, a traveller can faintly hear the
striking of the theatre clock. Finally the smell of _kira_ flowers fills
the breeze, and one has truly arrived.

Near the dancing-hall of Rizhti, the back-streets are lined with
_lirbsha_ trees, and the blue petals sway in the air. A distracted
visitor will miss the sandstone carvings which decorate the rooftops.
These commemorate the pestilence which recently devastated a nearby
village. The traveller may usually hear the delicate song of the _raski_
birds. The idle rich teem here, laughing and dancing.

# Tubil

Leaving Gulztig one can journey to Tubil. It is a smooth journey. As a
traveller enters Chalszhi, there is a change in the light, and a grey
cast falls over the land. Along the way the track crosses a broad
valley. By the road, hairy _ruzhsir_ graze.

At the outskirts of Tubil, one can espy the iniquitous mint of Zhubar.
From so far, the traveller has no way to observe how actors steer clear
of it. Closer to the centre of the city, the traveller begins to hear
songs of sorrowful delight, sung by petty criminals of the surrounding
countryside. Ultimately the perfume of _ritig_ fruit fills the air, and
a traveller has without a doubt arrived.

Close to the counting-house of Mizhbar, the streets are lined with
_raltarb_ trees, and the pleasant scent fills the late morning breeze.
Farmers congregate here, talking. A traveller will see the _zhatsal_
plants. These function as a caution to hostile influences.

# Rushib

Leaving Tubil one can journey west to Rushib. The route is long, but
rewarding, taking two days. On the verge of the road, shaggy _ruzhsir_
graze. Along the way the track crosses a broad valley, spotted with
_lirbsha_ plants. When the traveller enters Mabzirsh, there is a shift
in the light, and a pale cast falls over the landscape.

On entering the monumental city, the traveller may encounter goldsmiths,
hawking their wares. Slightly nearer to the inner parts of the city, a
traveller can faintly hear the clanging of the barbican clock.
Ultimately the bouquet of _kira_ flowers fills the breeze, and one is
undoubtedly in Rushib.

In the New Quarter of the city, the roadways are paved with stone,
inlaid with ivory. Soldiers throng here, singing and telling stories. An
inattentive observer will overlook the obsidian statues that grace the
eaves. These act as a relic of the war which long ago beset a nearby
village.

**Comment by ikarth** [comment]
I love it! Though it is making me feel like I should hurry up if I want to try something clever with a travel journal.

**Comment by MichaelPaulukonis** [comment]
So, your map is displayed East to West, South to North?

Since traveling west from Shurgulz leads to Kirrab, and not the ocean.

Likewise, north of Kirrab lies Rutig, not the sleepy hamlet of Rubil.

---

Otherwise, I am liking the look and sound of this.

**Comment by mewo2** [comment]
Yes, the map is displayed with south at the top, as is cartographic tradition in these lands.

**Comment by hugovk** [comment]
> Yes, the map is displayed with south at the top, as is cartographic tradition in these lands.

Nice! How about a compass point on the map to make it more evident?

**Comment by mewo2** [comment]
I didn't want to make too big a deal about it, but I'll consider it.

**Comment by spc476** [comment]
Are you generating the maps randomly too?  What are you using to draw them?

**Comment by erbridge** [comment]
I would love to see the source for these. They're really nice.

**Comment by mewo2** [comment]
I'm a little late with posting this, because I'm travelling right now. Code and final novel are now available.

Code: https://github.com/mewo2/deserts
Final novel: https://github.com/mewo2/deserts/raw/master/novel.pdf

**Comment by hugovk** [comment]
It's good!

> I'm a little late with posting this, because I'm travelling right now.

Apt!

**Comment by mewo2** [comment]
Now available in Twitter bot form! https://twitter.com/unchartedatlas

**Comment by hugovk** [comment]
:+1: Bot all the books!

**Comment by mewo2** [comment]
I wrote an explanation of the placename generator: http://mewo2.com/notes/naming-language/

And an explanation of the map generator: http://mewo2.com/notes/terrain/
